contracts/Bridge/BaseXVSProxyOFT.sol

```diff
@@ -0,0 +1,275 @@
+    }
+
+    /**
+     * @notice Sets the whitelist address to skip checks on transaction limit.
+     * @param user_ Adress to be add in whitelist.
+     * @param val_ Boolean to be set (true for isWhitelisted address)
+     */
+    function setWhitelist(address user_, bool val_) external onlyOwner {
+        emit SetWhitelist(user_, val_);
+        whitelist[user_] = val_;
+    }
+
+    /**
+     * @notice Triggers stopped state of the bridge.
+     */
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    /**
+     * @notice Triggers resume state of the bridge.
+     */
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    /**
+     * @notice Empty implementation of renounce ownership to avoid any mishappening.
+     */
+    function renounceOwnership() public override {}
+
+    function token() public view override returns (address) {
+        return address(innerToken);
+    }
+
+    function _isEligibleToSend(address from_, uint16 dstChainId_, uint256 amount_) internal {
+        // Check if the recipient's address is whitelisted
+        bool isWhiteListedUser = whitelist[from_];
+
+        // Calculate the amount in USD using the oracle price
+        uint256 amountInUsd;
+        Exp memory oraclePrice = Exp({ mantissa: oracle.getPrice(token()) });
+        amountInUsd = mul_ScalarTruncate(oraclePrice, amount_);
+
+        // Load values for the 24-hour window checks
+        uint256 currentBlockTimestamp = block.timestamp;
+        uint256 lastDayWindowStart = chainIdToLast24HourWindowStart[dstChainId_];
+        uint256 transferredInWindow = chainIdToLast24HourTransferred[dstChainId_];
+        uint256 maxSingleTransactionLimit = chainIdToMaxSingleTransactionLimit[dstChainId_];
+        uint256 maxDailyLimit = chainIdToMaxDailyLimit[dstChainId_];
+
+        // Revert if the amount exceeds the single transaction limit and the recipient is not whitelisted
+        require(amountInUsd <= maxSingleTransactionLimit || isWhiteListedUser, "Single Transaction Limit Exceed");
+
+        // Check if the time window has changed (more than 24 hours have passed)
+        if (currentBlockTimestamp - lastDayWindowStart > 1 days) {
+            transferredInWindow = amountInUsd;
+            chainIdToLast24HourWindowStart[dstChainId_] = currentBlockTimestamp;
+        } else {
+            transferredInWindow += amountInUsd;
+        }
+
+        // Revert if the amount exceeds the daily limit and the recipient is not whitelisted
+        require(transferredInWindow <= maxDailyLimit || isWhiteListedUser, "Daily Transaction Limit Exceed");
+
+        // Update the amount for the 24-hour window
+        chainIdToLast24HourTransferred[dstChainId_] = transferredInWindow;
+    }
+
+    function _isEligibleToReceive(uint16 srcChainId_, uint256 receivedAmount_) internal {
+        // Check if the sender's address is whitelisted
+        bool isWhiteListedUser = whitelist[msg.sender];
```

Comment:
Will not be the sender always the LZ relayer? Shouldn't we provide the `to` address as a parameter and consider it in the checks?